server/test/connected_objects.go

```diff
@@ -0,0 +1,670 @@
+							"parent": {Userset: &openfgapb.Userset_This{}},
+							"viewer": {
+								Userset: &openfgapb.Userset_Union{
+									Union: &openfgapb.Usersets{
+										Child: []*openfgapb.Userset{
+											{Userset: &openfgapb.Userset_This{}},
+											{Userset: &openfgapb.Userset_TupleToUserset{
+												TupleToUserset: &openfgapb.TupleToUserset{
+													Tupleset:        &openfgapb.ObjectRelation{Relation: "parent"},
+													ComputedUserset: &openfgapb.ObjectRelation{Relation: "viewer"},
+												},
+											}},
+										},
+									},
+								},
+							},
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationMetadata{
+								"parent": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{Type: "folder"},
+									},
+								},
+								"viewer": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{Type: "user"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			tuples:          []*openfgapb.TupleKey{},
+			expectedObjects: []string{"folder:folder1", "folder:folderX", "folder:folderY"},
+		},
+		{
+			name: "Restrict results based on limit",
+			request: &commands.ConnectedObjectsRequest{
+				StoreID:          ulid.Make().String(),
+				ObjectType:       "folder",
+				Relation:         "viewer",
+				User:             &openfgapb.ObjectRelation{Object: "user:jon"},
+				ContextualTuples: []*openfgapb.TupleKey{},
+			},
+			limit: 1,
+			model: &openfgapb.AuthorizationModel{
+				TypeDefinitions: []*openfgapb.TypeDefinition{
+					{
+						Type: "user",
+					},
+					{
+						Type: "folder",
+						Relations: map[string]*openfgapb.Userset{
+							"parent": {Userset: &openfgapb.Userset_This{}},
+							"viewer": {
+								Userset: &openfgapb.Userset_Union{
+									Union: &openfgapb.Usersets{
+										Child: []*openfgapb.Userset{
+											{Userset: &openfgapb.Userset_This{}},
+											{Userset: &openfgapb.Userset_TupleToUserset{
+												TupleToUserset: &openfgapb.TupleToUserset{
+													Tupleset:        &openfgapb.ObjectRelation{Relation: "parent"},
+													ComputedUserset: &openfgapb.ObjectRelation{Relation: "viewer"},
+												},
+											}},
+										},
+									},
+								},
+							},
+						},
+						Metadata: &openfgapb.Metadata{
+							Relations: map[string]*openfgapb.RelationMetadata{
+								"parent": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{Type: "folder"},
+									},
+								},
+								"viewer": {
+									DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+										{Type: "user"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			tuples: []*openfgapb.TupleKey{
+				tuple.NewTupleKey("folder:folder1", "viewer", "user:jon"),
+				tuple.NewTupleKey("folder:folderX", "parent", "*"),
+			},
+			expectedObjects: []string{"folder:folder1"},
```

Comment:
Technically this test case needs to be changed and some changes in ConnectedObjects as well, because with our recent changes the `folder:folderX#parent@*` tuple should lead to an error. Wildcards aren't allowed on tupleset relations.

Reply:
I've refactored the ConnectedObjects API a bit so that it returns the same error as the Check path if a `*` is encountered on a tupleset relation (e.g. `folder#parent`).

Let me know what you think 👍